src/aspire/volume/__init__.py

```diff
@@ -287,6 +358,91 @@ class FBBasisVolume(BasisVolume):
     pass
 
 
+def gaussian_blob_Cn_vols(
+    L=8, C=2, K=16, alpha=1, subscript=1, seed=None, dtype=np.float64
+):
+    """
+    Generate Cn rotationally symmetric volumes composed of Gaussian blobs.
+    The volumes are symmetric about the z-axis.
+
+    Defaults to volumes with no symmetry.
+
+    :param L: The size of the volumes
+    :param C: The number of volumes to generate
+    :param K: The number of blobs each volume is composed of.
+    A Cn symmetric volume will be composed of n times K blobs.
+    :param subscript: The subcript n indicating Cn symmetry.
+    :param alpha: A scale factor of the blob widths
+
+    :return: A Volume instance containing C Gaussian blob volumes with Cn symmetry.
+    """
+
+    def _eval_gaussian_blobs(L, Q, D, mu, subscript, dtype=np.float64):
+        try:
+            subscript = int(subscript)
+        except Exception:
+            raise NotImplementedError(
+                f"C{subscript} symmetry not supported. Only Cn symmetry, where n is an integer, is supported."
+            )
+
+        g = grid_3d(L, dtype=dtype)
+        coords = np.array(
+            [g["x"].flatten(), g["y"].flatten(), g["z"].flatten()], dtype=dtype
+        )
+
+        K = Q.shape[-1]
+        vol = np.zeros(shape=(1, coords.shape[-1])).astype(dtype)
+        rot = np.zeros(shape=(3, 3, subscript)).astype(dtype)
+
+        for k in range(subscript):
+            rot[:, :, k] = [
+                [
+                    np.cos(2 * np.pi * k / subscript),
+                    -np.sin(2 * np.pi * k / subscript),
+                    0,
+                ],
+                [
+                    np.sin(2 * np.pi * k / subscript),
+                    np.cos(2 * np.pi * k / subscript),
+                    0,
+                ],
+                [0, 0, 1],
+            ]
+
+        for k in range(K):
+            for j in range(subscript):
+                coords_k = rot[:, :, j] @ coords - mu[:, k, np.newaxis]
+                coords_k = (
+                    Q[:, :, k] / np.sqrt(np.diag(D[:, :, k])) @ Q[:, :, k].T @ coords_k
```

Comment:
It does work if we drop the second `Q[:, :, k]`, but not the first. I believe the form `(Q*Q.T)/sqrt(D)` is close to correct. The `Q*Q.T` is the quadratic term of centers/means in the gaussian function and the  `sqrt(D)` appears to be the square root of the variances (not sure we should be taking the square root here). 

At any rate, this was all existing code for the original gaussian volume generator. I can dig in a little more to see if this is a correct formulation.

Reply:
Ok, this term caused me quite a bit of confusion when going through the original gaussian blob code. It looks like we can drop the second `Q[:, :, k]` and everything still works. Dropping the first `Q[:, :, k]` zeros out the volume for some reason. Maybe we can discuss in dev meeting.

Reply:
Replaced the first `Q[:, :, k]` with an identity matrix.

Reply:
Could probably get rid of it completely since we just have a broadcasting elementwise division here. May have to move the division to after the matrix product, though.

Reply:
We're unable to move this after the matrix product because `coords_k` has shape 3 x L^3. Moving it behind `Q` causes the division to broadcast along the wrong array axis (this would apply the variances incorrectly across coordinates). I'm not 100% on this, but I think we need to leave it where it is.

Reply:
Was able to move behind the matrix product by using `np.newaxis`.